sympy/series/formal.py

```diff
@@ -1143,6 +1143,66 @@ def integrate(self, x=None, **kwargs):
 
         return self.func(f, self.x, self.x0, self.dir, (ak, self.xk, ind))
 
+    def convolve(self, other, x=None, order=4):
+        """ Convolute two Formal Power Series and return the truncated terms upto specified order.
+
+        Parameters
+        ==========
+
+        order : Number, optional
+            Specifies the order of the term upto which the polynomial should
+            be truncated.
+
+        Examples
+        ========
+
+        >>> from sympy import fps, sin, exp, convolution
+        >>> from sympy.abc import x
+        >>> f1 = fps(sin(x))
+        >>> f2 = fps(exp(x))
+
+        >>> f1.convolve(f2, x, 6)
+        x + x**2 + x**3/3 - x**5/12 + O(x**6)
+
+        See Also
+        ========
+
+        sympy.discrete.convolutions
+        """
+        from sympy.discrete.convolutions import convolution
```

Comment:
Can this be top level import?

Reply:
Just asking, can you please clarify on when the import should be top level and when not? Some conditions for this?

Reply:
> Just asking, can you please clarify on when the import should be top level and when not? Some conditions for this?

We try to follow the convention of imports at top-level only. It's easier to debug and read since you can see a list of everything up-top.

The only time I have ever added these imports inside of functions is when there are errors due to cyclic imports.